src/modules/sub/crypto.js

```diff
@@ -1,5 +1,26 @@
-import { createHmac } from "crypto";
+import { createHmac, createCipheriv, createDecipheriv, scryptSync } from "crypto";
+
+const algorithm = "aes256"
+const keyLength = 32;
 
 export function sha256(str, salt) {
-    return createHmac("sha256", salt).update(str).digest("hex");
+    return createHmac("sha256", salt).update(str).digest("base64");
+}
+
+export function encryptStream(str, iv, secret) {
+    const buff = Buffer.from(JSON.stringify(str), "utf-8");
+
+    const key = scryptSync(Buffer.from(secret, "base64"), "salt", keyLength);
+    const cipher = createCipheriv(algorithm, key, Buffer.from(iv, "base64"));
+
+    return Buffer.from(cipher.update(buff, "utf8", "binary") + cipher.final("binary"), "binary");
```

Comment:
```js
    return Buffer.concat([ cipher.update(buff), cipher.final() ])
```
same below

Reply:
it would also be more descriptive if `buff` was `plaintext` (and `ciphertext` in decrypt)